UnityProject/Assets/Scripts/Input System/Triggers/InputTrigger.cs

```diff
@@ -106,5 +106,32 @@ public bool Trigger(Vector3 position)
 	public virtual void UI_Interact(GameObject originator, string hand) {}
 	public virtual bool UI_InteractOtherSlot(GameObject originator, GameObject otherHandItem){ return true; }
 
+    /// <Summary>
+    /// This is called by the Interact() of objects twice, one by the client and then by the server if the first call returns true
+    /// </Summary>
+    /// <param name="originator">game object that is performing the interaction upon this gameobject</param>
+    /// <param name="hand">hand of the originator which is being used to perform the interaction</param>
+    /// <param name="position">position of the interaction</param>
+	/// <param name="allowSoftCrit">if true it allows mobs in soft crit perform the action</param>
+    /// <returns>true if the server confirms the interaction is valid. The Interact() following this will be carried on serverside</returns>
+    public bool canUse(GameObject originator, string hand, Vector3 position, bool allowSoftCrit){

```

Comment:
I'd set a default value for softcrit, also please rename method to PascalCase, it's not java
```suggestion
    public bool CanUse(GameObject originator, string hand, Vector3 position, bool allowSoftCrit = false){
```